src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,83 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { getUserPosts } from '../../api/posts';
 import './PostsList.scss';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
+interface Post {
+  id: number,
+  userId: number,
+  title: string,
+}
 
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+type Props = {
+  selectedUser: string,
+  selectPost: (postId: number) => void,
+  selectedPostId: number,
+  getDetails: (id: number) => void,
+  removeDetails: () => void,
+};
 
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
+export const PostsList: React.FC<Props> = ({
+  selectedUser,
+  selectPost,
+  selectedPostId,
+  getDetails,
+  removeDetails,
+}) => {
+  const [posts, setPosts] = useState([]);
 
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
-  </div>
-);
+  useEffect(() => {
+    getUserPosts(selectedUser).then(listOfPosts => {
+      setPosts(listOfPosts);
+    });
+  });
+
+  return (
+    <div className="PostsList">
+      <h2>Posts:</h2>
+
+      <ul className="PostsList__list">
+        {posts.length > 0 && posts.map((post: Post) => {
```

Comment:
```suggestion
        {posts.map((post: Post) => {
```
если длина ноль, то просто ничего не выведется